File: common/src/main/java/org/cleanas2/common/ConnectionInfo.java
package org.cleanas2.common;

import org.apache.http.HttpRequest;
import org.apache.http.impl.BHttpConnectionBase;

/**
 * Contains information about an incoming connection.  used in handling incoming messages, and in debugging.
 *
 * @author Andrew Backer {@literal awbacker@gmail.com / andrew.backer@powere2e.com}
 */
public class ConnectionInfo {
    public String sourceIp = "";
    public int sourcePort = 0;
    public String destinationIp = "";
    public int destinationPort = 0;
    public String requestMethod = "";
    public String requestUri = "";

    public ConnectionInfo() {
    }

    public ConnectionInfo(BHttpConnectionBase connection, HttpRequest request) {
        populateFrom(connection, request);
    }

    void populateFrom(BHttpConnectionBase connection, HttpRequest request) {
        sourceIp = connection.getRemoteAddress().toString();
        sourcePort = connection.getLocalPort();
        destinationIp = connection.getLocalAddress().toString();
        destinationPort = connection.getLocalPort();
        requestMethod = request.getRequestLine().getMethod();
        requestUri = request.getRequestLine().getUri();
    }
}

File: common/src/main/java/org/cleanas2/common/ConnectionInfo.kt
package org.cleanas2.common

import org.apache.http.HttpRequest
import org.apache.http.impl.BHttpConnectionBase

/**
 * Contains information about an incoming connection.  used in handling incoming messages, and in debugging.
 *
 * @author Andrew Backer awbacker@gmail.com / andrew.backer@powere2e.com
 */
class ConnectionInfo {
    var sourceIp = ""
    var sourcePort = 0
    var destinationIp = ""
    var destinationPort = 0
    var requestMethod = ""
    var requestUri = ""

    constructor() {}

    constructor(connection: BHttpConnectionBase, request: HttpRequest) {
        populateFrom(connection, request)
    }

    internal fun populateFrom(connection: BHttpConnectionBase, request: HttpRequest) {
        sourceIp = connection.remoteAddress.toString()
        sourcePort = connection.localPort
        destinationIp = connection.localAddress.toString()
        destinationPort = connection.localPort
        requestMethod = request.requestLine.method
        requestUri = request.requestLine.uri
    }
}
